### PR TITLE
[border-agent] log enhancements

### PR DESCRIPTION
**Pull request by librasungirl**
This commit updates logs on secure session establishment and commissioner for accuracy.

### CODE REVIEW

**Comment by bukepo** [comment]
The timeout is useful to detect and tear down stale session. Should we keep it?

File: src/core/meshcop/border_agent.cpp [hunk 1]
```diff
@@ -447,6 +447,7 @@ template <>
 void BorderAgent::HandleTmf<kUriCommissionerPetition>(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
 {
     IgnoreError(ForwardToLeader(aMessage, aMessageInfo, kUriLeaderPetition));
+    mTimer.Start(kKeepAliveTimeout);
```
Comment (abtink) [inline src/core/meshcop/border_agent.cpp:450]: This now starts the timer when the BA forwards the "petition".

Should we also remove the starting of timer in `BorderAgent::HandleConnected()`?

**Comment by librasungirl** [comment]
> The timeout is useful to detect and tear down stale session. Should we keep it?

Thanks for sharing the valid concern. There is indeed risk that BA unable to serve for stale session when commissioner candidate not gracefully disconnect. 

Will file TG jira to ask for clarification on whether or not the kKeepAliveTimeout is applicable to secure session for candidate and how to keep alive for candidate if needed.

Here will update this PR for log enhancements only.

File: src/core/meshcop/border_agent.cpp (hunk 1 above)
Comment (librasungirl) [reply src/core/meshcop/border_agent.cpp]: Rescoped this PR for log enhancements only for the discussion https://github.com/openthread/openthread/pull/10549#issuecomment-2264930711

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**